acra/guides/advanced-integrations/client-side-integration-with-acra-connector.md

```diff
@@ -0,0 +1,224 @@
+
+#### Why do we need a special piece of software to talk to your other piece of software?
+
+Acra needs a trusted agent on the application's side to protect the sensitive decrypted responses, to provide basic channel authentication, and to enforce certain behaviours.
+
+AcraConnector acts as a local proxy that receives requests from the app and returns decrypted answers. AcraConnector provides an encrypted and authenticated connection to AcraServer, which, in turn, fetches the response to the request from the database and decrypts all the data. AcraServer then returns the data to AcraConnector via a secure connection.
+
+AcraConnector works in a similar fashion with AcraTranslator, redirecting AcraStructs from application to AcraTranslator and delivering the decrypted response back.
+
+This enforces maximum secrecy, employing authentication that is easy to manage (pre-shared private keys), and requires minimal intervention into your code for successful implementation!
+
+
+### Getting started with AcraConnector
+
+#### Method 1A. Launch AcraConnector using Docker (the fastest way to try AcraConnector)
+> Note: Using Docker is recommended for testing purposes only. Please don't rely on Docker in real-life production settings.    
+> Note: The following examples focus on using AcraConnector and AcraWriter with PostgreSQL, but Acra also supports MySQL.
+
+Clone the Acra repository, build images, and start Docker compose with PostgreSQL, AcraServer, AcraConnector, and Secure Session between them:
+
+```
+git clone https://github.com/cossacklabs/acra.git
+make docker
+docker-compose -f docker-compose.pgsql-nossl-server-ssession-connector.yml up
+```
+
+After running this command, the basic infrastructure is all set up, all components are connected, and the keys are distributed into the appropriate folder.
+
+Now, proceed to step [launching AcraConnector]({{< ref "acra/configuring-maintaining/general-configuration/acra-connector.md#launching-acraconnector-INVALID" >}}).
+Or install AcraConnector manually.
+
+#### Method 1B. Manual launch
+
+> Note: Skip this if you've used the Docker method described above.
+
+- Install dependencies: [Themis]({{< ref "/themis/installation/" >}}) cryptographic library:
+
+```
+git clone https://github.com/cossacklabs/themis.git
+cd themis
+make
+sudo make install
+```
+
+- Install AcraConnector:
+
+```
+go get github.com/cossacklabs/acra/cmd/acra-connector
+```
+
+> Note: Hereinafter all the commands starting with 'go' are meant to be executed from the 'acra' folder (the folder with the repository code).
+
+- Install key generation utility:
+
+```
+go get github.com/cossacklabs/acra/cmd/acra-keymaker
+```
+
+- Use `acra-keymaker` to generate master key into `master.key` file and assign it into the environment variable like this:
+```
+$GOPATH/bin/acra-keymaker --generate_master_key=master.key
```

Comment:
Yes and no. From `go help install`:

```
Executables are installed in the directory named by the GOBIN environment
variable, which defaults to $GOPATH/bin or $HOME/go/bin if the GOPATH
environment variable is not set. Executables in $GOROOT
are installed in $GOROOT/bin or $GOTOOLDIR instead of $GOBIN.
```
In the last golang versions `GOPATH` is not required for `go` calls. So users may define or omit it. But in our documentation of AcraWriter installation for golang we set `GOPATH` to `$HOME/work`. Anyway, I will change it to $GOBIN and add note and link to golang documentation to help users find executables in their environment